examples/generative/speaker_recognition_using_cnn.py

```diff
@@ -0,0 +1,520 @@
+"""
+Title: Speaker Recognition Using a Convolutional Network
+Author: Fadi Badine
+Date created: 14/06/2020
+Last modified: 23/06/2020
+Description: A CNN to classify speakers using Fast Fourier Transformation.
+"""
+"""
+## Introduction
+
+This example demostrates how to create a model to classify speakers based on the
+transformation of their speech into the frequency domain using Fast Fourier
+Transformation.
+
+It shows the following:
+- How to use tf.data to load, preprocess and feed audio stream into the model
+- How to create a convolutional model using Conv1D and Max1Pool along with residual
+connections.
+
+Note:
+- This example should be run with `TensorFlow 2.3` or higher, or `tf-nightly`.
+- The noise samples in the dataset need to be resampled to a sampling rate of 16000
+before using the code in this example
+"""
+
+"""
+## Setup
+"""
+
+import os
+import shutil
+import numpy as np
+
+import tensorflow as tf
+from tensorflow import keras
+
+from tensorflow.python.keras.preprocessing import dataset_utils
+
+# Where the dataset is located
+# Get the data from https://www.kaggle.com/kongaevans/speaker-recognition-dataset/download
+DATASET_ROOT = "/home/fadi/Downloads/16000_pcm_speeches"
```

Comment:
Prefer using `HOME + /Downloads` so that others will be able to run the code without modifying the Python file, just by downloading the data

Reply:
Done